federation-js/src/composition/utils.ts

```diff
@@ -62,20 +62,39 @@ export function mapFieldNamesToServiceName<Node extends { name: NameNode }>(
 
 export function findDirectivesOnNode(
   node: Maybe<
-    | TypeDefinitionNode
-    | TypeExtensionNode
     | FieldDefinitionNode
-    | SchemaDefinitionNode
+    | FieldNode
+    | DefinitionNode
```

Comment:
I like the direction you're taking this. I will suggest, I think we should either go full bore `Maybe<ASTNode>` as the type or only permit types which at least have a `directives` property on them. I like the latter, since the typings will warn users of the fn when they're using it inappropriately. In your case, the only inappropriate usage which is now included in this union is the `DirectiveDefinitonNode`, so I'll suggest this (+ a nice little TS revamp since we're here)

```ts
export function findDirectivesOnNode(
  node: Maybe<
    Exclude<
      | FieldDefinitionNode
      | InputValueDefinitionNode
      | FieldNode
      | DefinitionNode,
      DirectiveDefinitionNode
    >
  >,
  directiveName: string,
) {
  return (
    node?.directives?.filter(
      (directive) => directive.name.value === directiveName,
    ) ?? []
  );
}
```